File: 1099_two_sum_less_than_k/two_sum_less_than_k.java
class Solution {
    // Time: O(n * log(n))
    // Space: O(1)
    public int twoSumLessThanK(int[] A, int K) {
        int res = -1;
        Arrays.sort(A);
        int i = 0, j = A.length - 1;
        while (i < j) {
            if (A[i] + A[j] < K) {
                res = Math.max(res, A[i] + A[j]);
                i++;
            } else {
                j--;
            }
        }
        return res;
    }
}

File: 1099_two_sum_less_than_k/two_sum_less_than_k.py
from typing import List


def twoSumLessThanK(A: List[int], K: int) -> int:
    """
    Time: O(n * log(n))
    Space: O(1)
    """
    res = -1
    A.sort()
    lo, hi = 0, len(A) -1
    while lo < hi:
        if (A[lo] + A[hi] < K):
            res = max(res, A[lo] + A[hi])
            lo += 1
        else:
            hi -= 1
    return res